zio-kafka/src/main/scala/zio/kafka/consumer/internal/Runloop.scala

```diff
@@ -560,6 +560,17 @@ private[consumer] final class Runloop private (
                                          ended = endedStreams.map(_.tp).toSet
                                        )
                                      )
+                                // Ensure that all assigned partitions have a stream and no streams are present for unassigned streams
+                                _ <-
+                                  ZIO
+                                    .logWarning(
+                                      s"Not all assigned partitions have a (single) stream or vice versa. Assigned: ${currentAssigned.mkString(",")}, streams: ${updatedAssignedStreams.map(_.tp).mkString(",")}"
+                                    )
+                                    .when(
+                                      currentAssigned != updatedAssignedStreams
+                                        .map(_.tp)
+                                        .toSet || currentAssigned.size != updatedAssignedStreams.size
```

Comment:
This can be simplified to:
```suggestion
                                      currentAssigned != updatedAssignedStreams.map(_.tp).toSet
```
although we could do this as a (premature) performance optimization:
```suggestion
                                      currentAssigned.size != updatedAssignedStreams.size ||
                                         currentAssigned != updatedAssignedStreams.map(_.tp).toSet
```

Reply:
I wanted to make sure that there are no duplicate TPs in `assignedStreams`, hence the size check as well. Also the size check is probably faster as the first condition to check.

Reply:
Of course, you are right, `updatedAssignedStreams` is not a set! 👍